src/tcpconns.c

```diff
@@ -486,6 +494,133 @@ static int conn_handle_ports (uint16_t port_local, uint16_t port_remote, uint8_t
 } /* int conn_handle_ports */
 
 #if KERNEL_LINUX
+
+#if HAVE_STRUCT_LINUX_INET_DIAG_REQ
+/* Batched reporting of value_list_t  */
+typedef struct value_list_batch_s {
+    size_t size;
+    size_t cur;
+    value_list_t buffer[1];       /* Allocated to have size entries */
+} value_list_batch_t;
+
+static value_list_batch_t *value_list_batch_create(size_t size)
+{
+    value_list_batch_t *ret =
+        malloc(sizeof(value_list_batch_t) + (size - 1) * sizeof(value_list_t));
+    size_t i;
+    const value_list_t vl_init = VALUE_LIST_INIT;
+    if (size < 4) {
+        ERROR("Buffer size %zu < 4; using 4", size);
+        size = 4;
+    }
+    ret->size = size;
+    for (i = 0; i < size; i++)
+        ret->buffer[i] = vl_init;
+    ret->cur = 0;
+    return ret;
+}
+
+static void value_list_batch_flush(value_list_batch_t *batch)
+{
+    size_t i;
+    /* TODO(arielshaqed): Use new batched reporting API! */
+    for (i = 0; i < batch->cur; i++) {
+        plugin_dispatch_values (&batch->buffer[i]);
+    }
+    for (i = 0; i < batch->cur; i++) {
+        free(batch->buffer[i].values);
+        /* Also free meta? */
```

Comment:
`plugin_dispatch_values()` gets a `const` pointer, so it will not allocate a meta object that you would have to free here.